### ISSUE TITLE
pgnodemx: expected 2 tokens, found 3

### ISSUE THREAD

**Issue by marciomed1** [CrunchyData/pgnodemx#7]
I get the following error when quering the function kdapi_setof_kv('annotations') of pgnodemx:

pgnodemx: incorrect format for key equals quoted value line
  Detail: pgnodemx: expected 2 tokens, found 3

The offending values contains an equal symbol '=', even thought they are quoted. The values comes from an annotated PGO v5.0.3 Centos8 cluster like the following example:

```
instances:
    metadata:
      annotations:
        pwgen: cHdnZW4=
```

This becomes the following for the pgnodemx extension:

```
pwgen="cHdnZW4="
```

Because the values from K8s are often base64 encoded, it's not rare to get an equal symbol in them.

The function [`parse_keqv_line()`](https://github.com/CrunchyData/pgnodemx/blob/master/parseutils.c#L404) seems to not take in account the quoted string as a whole, but instead keeps splitting the string by equal symbol inside the quoted text.

Regards,

**Comment by pgguru** [comment]
I've been able to reproduce and verify that the underlying file that it reads (`/etc/podinfo/annotations`) is formatted correctly as you'd expect:

```
kubernetes.io/config.seen="2021-11-18T18:19:12.917753588Z"
kubernetes.io/config.source="api"
pwgen="abasdfd=="
status="{\"conn_url\":\"postgres://hippo-instance1-h68h-0.hippo-pods:5432/postgres\",\"api_url\":\"https://hippo-instance1-h68h-0.hippo-pods:8008/patroni\",\"state\":\"running\",\"role\":\"replica\",\"version\":\"2.1.1\",\"xlog_location\":103425760,\"timeline\":1}"
```

In looking at this, looks like the parsing routine needs some additional smarts added; in addition, I think we should verify that it's handling internal quotes/backslashes correctly; from comments, it looks like it's just stripping leading/trailing quotes, which wouldn't be sufficient in this case.

**Comment by pgguru** [comment]
Just playing around additionally with parsing here; looks like annotations accept newlines, which get rendered thusly in `/etc/podinfo/annotations`:

```
multiline="foo\n\nbar"
```

Would be useful to see if there is some formal specification of the serialization in use, as we would (presumably) want to return the deserialized/original annotated value.  Thoughts @jkatz or @jconway if this is in fact the expected behavior (i.e., annotated value input = function output)?

**Comment by pgguru** [comment]
With PR #8, the following (mocked) `/etc/podinfo/annotations`:
```
.io/config.seen="2021-11-18T18:19:12.917753588Z"
kubernetes.io/config.source="api"
pwgen="abasdfd=="
status="{\"conn_url\":\"postgres://hippo-instance1-h68h-0.hippo-pods:5432/postgres\",\"api_url\":\"https://hippo-instance1-h68h-0.hippo-pods:8008/patroni\",\"state\":\"running\",\"role\":\"replica\",\"version\":\"2.1.1\",\"xlog_location\":103425760,\"timeline\":1}"
```

returns the following:

```
             key             |                                                                                                                   val
-----------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 .io/config.seen             | 2021-11-18T18:19:12.917753588Z
 kubernetes.io/config.source | api
 pwgen                       | abasdfd==
 status                      | {"conn_url":"postgres://hippo-instance1-h68h-0.hippo-pods:5432/postgres","api_url":"https://hippo-instance1-h68h-0.hippo-pods:8008/patroni","state":"running","role":"replica","version":"2.1.1","xlog_location":103425760,"timeline":1}
(4 rows)
```

**Comment by pgguru** [comment]
Also added full parsing/handling of Go's quoted string format, as this is the origination of this source.  Now with emojis! :)

**Comment by pgguru** [comment]
Fixed in #8.
